core/src/main/scala/chisel3/experimental/verification/package.scala

```diff
@@ -3,41 +3,55 @@
 package chisel3.experimental
 
 import chisel3._
-import chisel3.internal.Builder
+import chisel3.internal.{Builder, NamedComponent}
 import chisel3.internal.firrtl.{Formal, Verification}
 import chisel3.internal.sourceinfo.SourceInfo
 
 package object verification {
+
+  /** Base simulation-only component. */
+  abstract class BaseSim extends NamedComponent
+
+  case class Assert(predicate: Bool, msg: String = "") extends BaseSim
+  case class Assume(predicate: Bool, msg: String = "") extends BaseSim
+  case class Cover(predicate: Bool, msg: String = "") extends BaseSim
```

Comment:
These should not be `case class`es because all `Data` (`Bool` included) implement referential equality and are thus not appropriate for the structural equality implications of `case classes`.

They should be `final class`es though to ensure they are not user extensible by accident.